plugins/mac-capture/mac-screen-capture.m

```diff
@@ -317,19 +313,14 @@ static bool init_screen_stream(struct screen_capture *sc)
             SCDisplay *target_display = get_target_display();
 
             if (sc->hide_obs) {
-                __block SCRunningApplication *obsApp = nil;
-                [sc->shareable_content.applications
-                    indexOfObjectPassingTest:^BOOL(SCRunningApplication *_Nonnull app, NSUInteger idx __unused,
-                                                   BOOL *_Nonnull stop) {
-                        if ([app.bundleIdentifier isEqualToString:[[NSBundle mainBundle] bundleIdentifier]]) {
-                            obsApp = app;
-                            *stop = TRUE;
-                        }
-                        return *stop;
-                    }];
-
-                NSArray *exclusions = [[NSArray alloc] initWithObjects:obsApp, nil];
-
+                NSMutableArray *exclusions = [[NSMutableArray alloc] init];
```

Comment:
For now, exclusions at most will contain a single element (namely OBS Studio itself), so using a mutable array seems wasteful to me.

I'd rather declare an `NSArray` and init it with the single element inside the loop.

Reply:
This is in fact intentional change. Logically, there is the case that we could have a chance that we don't have application that has bundle id that same as the main bundle.
In this case, in the original code we may have `nil` in `NSArray` which crashes the app. (`NSArray` can't hold `nil` in its item.) If we want to safely use `NSArray` and if it's not found use empty array, we can write such code, but it would be more complicated.
The cost of `NSMutableArray` is not really high and ignoreable in this use case I think.
How do you think?

Reply:
An, never mind actaully, `initWithObjects:nil, nil` may not crash the app because it's `nil` ended, uh, but still tricky code for sure.

Reply:
This code only affects OBS. OBS identifies itself through its main bundle, and hence as long as it's running its bundle identifier in SCKs application list will be identical to its main bundle ID.

I can't think of a generation where that bundle ID is _not_ the main bundle ID.

Reply:
Yeah, I agree with that point. I said logically... practically, it must be always found. Anways, reverthing to use `NSArray` instead now. As I found set `nil` is safe for `initWithObjects:`, I don't have strong opinion on this change anways. Thank you for the comments.

Reply:
Updated.

Reply:
Right, logically it could happen and the NSArray reference would be `nil` because it's never assigned an allocated NSArray object.

I guess I'm fine with keeping the mutable type, as it also makes the code probably easier to read.

Reply:
Let's keep `NSArray` approach here for now, I agree that it's not necessary to use `NSMutableArray` for sure.